openeogeotrellis/job_tracker_v2.py

```diff
@@ -8,14 +8,13 @@
 import collections
 import datetime as dt
 import logging
-import re
-import subprocess
 from decimal import Decimal
 from pathlib import Path
-from subprocess import CalledProcessError
-from typing import Callable, List, NamedTuple, Optional, Union
+from typing import Any, List, NamedTuple, Optional, Union
 
 import requests
+import requests_gssapi
```

Comment:
I think we should make this an optional dependency under a "yarn" extra, 
because we don't need this on kube deploys

Reply:
I made the dependency optional as an extra requirement for yarn.

And here in job_tracker_v2 module I kept the import at the top of the module rather than doing a local import in the function that uses it. I think that makes it easier to keep a clear overview of what the module needs, and what it optionally uses.

Therefore it sets requests_gssapi to None if it is not available.

https://github.com/Open-EO/openeo-geopyspark-driver/blob/a8c943ada76f5ab4a414e6b7ec30570bf1e77295/openeogeotrellis/job_tracker_v2.py#L17-L21